front/src/applications/operationalStudies/hooks/useSetupItineraryForTrainUpdate.ts

```diff
@@ -88,20 +85,67 @@ export const useSetupItineraryForTrainUpdate = (
               const stepsCoordinates = pathfindingResult.path_items_positions.map((position) =>
                 getPointCoordinates(geometry, pathfindingResult.length, position)
               );
-
-              const formatedPathSteps: PathStep[] = trainSchedule.path.map((step, i) => ({
-                ...step,
-                coordinates: stepsCoordinates[i],
-                positionOnPath: pathfindingResult.path_items_positions[i],
-              }));
-
               const suggestedOperationalPoints: SuggestedOP[] = formatSuggestedOperationalPoints(
                 operational_points,
                 geometry,
                 pathfindingResult.length
               );
 
-              const allWaypoints = upsertViasInOPs(suggestedOperationalPoints, formatedPathSteps);
+              const updatedPathSteps: PathStep[] = trainSchedule.path.map((step, i) => {
+                const correspondingOp = suggestedOperationalPoints.find(
+                  (suggestedOp) =>
+                    'uic' in step &&
+                    suggestedOp.uic === step.uic &&
+                    suggestedOp.ch === step.secondary_code
+                );
+
+                const correspondingSchedule = trainSchedule.schedule?.find(
+                  (schedule) => schedule.at === step.id
+                );
+
+                const { kp, name, ch } = correspondingOp || {};
+
+                const {
+                  arrival,
+                  stop_for: stopFor,
+                  locked,
+                  on_stop_signal: onStopSignal,
+                } = correspondingSchedule || {};
+
+                const stepWithoutSecondaryCode =
+                  'secondary_code' in step ? omit(step, ['secondary_code']) : step;
+
+                return {
+                  ...stepWithoutSecondaryCode,
+                  ch,
+                  kp,
+                  name,
+                  positionOnPath: pathfindingResult.path_items_positions[i],
+                  arrival,
+                  stopFor,
```

Comment:
As the back send us the duration in ISO8601 (ex: "PTxxS"), we need to convert it back to seconds

```suggestion
                  stopFor: stopFor ? ISO8601Duration2sec(stopFor) : stopFor,
```